plotpal/volumes.py

```diff
@@ -0,0 +1,373 @@
+        
+        left_field = self._modify_field(left_field)
+        right_field = self._modify_field(right_field)
+        top_field = self._modify_field(top_field)
+        
+
+        left_mid=self.left_mid
+        if left_mid is not None:
+            mid_left_field = np.squeeze(dsets[self.left_mid][ni,:])
+            mid_right_field = np.squeeze(dsets[self.right_mid][ni,:])
+            mid_top_field = np.squeeze(dsets[self.top_mid][ni,:])
+            
+            if vector_ind is not None:
+                mid_left_field = mid_left_field[vector_ind,:]
+                mid_right_field = mid_right_field[vector_ind,:]
+                mid_top_field = mid_top_field[vector_ind,:]
+        
+            mid_left_field = self._modify_field(mid_left_field)
+            mid_right_field = self._modify_field(mid_right_field)
+            mid_top_field = self._modify_field(mid_top_field)
+            
+            xy_side = construct_surface_dict(self.x, self.y, self.Lz, top_field,x_bounds=(self.x_min, self.x_mid), y_bounds=(self.y_min, self.y_mid))
+            xz_side = construct_surface_dict(self.x, self.Ly, self.z, right_field, x_bounds=(self.x_min, self.x_mid), z_bounds=(self.z_min, self.z_mid))
+            yz_side = construct_surface_dict(self.Lx, self.y, self.z, left_field, y_bounds=(self.y_min, self.y_mid), z_bounds=(self.z_min, self.z_mid))
+            
+            xy_mid = construct_surface_dict(self.x, self.y, self.z_mid, mid_top_field,x_bounds=(self.x_mid, self.x_max), y_bounds=(self.y_mid, self.y_max), bool_function=np.logical_and)
+            xz_mid = construct_surface_dict(self.x, self.y_mid, self.z, mid_right_field, x_bounds=(self.x_mid, self.x_max), z_bounds=(self.z_mid, self.z_max), bool_function=np.logical_and)
+            yz_mid = construct_surface_dict(self.x_mid, self.y, self.z, mid_left_field, y_bounds=(self.y_mid, self.y_max), z_bounds=(self.z_mid, self.z_max), bool_function=np.logical_and)
+            
+        else:
+            xy_side = construct_surface_dict(self.x, self.y, self.Lz, top_field)
+            xz_side = construct_surface_dict(self.x, self.Ly, self.z, right_field)
+            yz_side = construct_surface_dict(self.Lx, self.y, self.z, left_field)
+            
+        cmap = matplotlib.cm.get_cmap(self.cmap)
+        vmin, vmax = self._get_minmax(left_field)
+        self.current_vmin, self.current_vmax = vmin, vmax
+        norm = matplotlib.colors.Normalize(vmin=vmin, vmax=vmax)
+        side_list = [xy_side, xz_side, yz_side]
+        if left_mid is not None:
+            side_list = [xy_side, xz_side, yz_side, xy_mid, xz_mid, yz_mid]
+        for d in side_list:
+            x = d['x']
+            y = d['y']
+            z = d['z']
+            sfc = cmap(norm(d['surfacecolor']))
+            surf = ax.plot_surface(x, y, z, facecolors=sfc, cstride=1, rstride=1, linewidth=0, antialiased=False, shade=False)
+            ax.plot_wireframe(x, y, z, ccount=1, rcount=1, linewidth=1, color='black')
```

Comment:
currently doesn't plot lines on the inside of the cutout

Reply:
TODO: (after our convo today) add wireframe outlines to the inside cutout. I think you only need to add to the vertical left and right slices (the horizontal slice will be filled in by all the other wireframes)